core/shared/src/main/scala/zio/FiberId.scala

```diff
@@ -31,6 +29,7 @@ sealed trait FiberId extends Serializable { self =>
 
   final def combine(that: FiberId): FiberId =
     (self, that) match {
+      case (None, None) => None
```

Comment:
Scala is super dumb about pattern matching. Instead of making a tuple and pattern matching on that, first match on the first element, and then match on the second. It's more painful but more performant.

e.g. instead of:

```scala
(a, b) match { case (a, b) => ... }
```

prefer:

```scala
a match { case a => b match { case b => ... } }
```

Reply:
Implemented